webapp/src/immersivereader.tsx

```diff
@@ -24,31 +30,44 @@ function beautifyText(content: string): string {
 
     return contentWIP;
 
-    // Change ``|| around blocks to {}
+    // Change ``|| around blocks to ""
     function cleanBlockAnnotation(content: string): string {
-        const blockLiteral = /``\|\|([\w|\s]+:[\w|\s]+)\|\|``/
-        const blockRegex = new RegExp(blockLiteral, "g");
-        const blocksRemoved = content.replace(blockRegex, (matched, word, offset, s) => {
-            return "{" + word + "}";
-        })
-        return blocksRemoved;
+        return content.replace(
+            /`?`\|\|[\w|\s]+:(.+?)\|\|``?/gu,
+            (matched, word, offset, s) => lf("\"{0}\"", word)
+        );
     }
 
     // Don't show any images
     function cleanImages(content: string): string {
-        const imageRegEx = /(!\[.*\]\(.*\))/g;
-        const imagesRemoved = content.replace(imageRegEx, "");
-        return imagesRemoved;
+        return content.replace(
+            /(!\w*\[[^\]]*\]\w*\([^)]*\))/gu,
+            ""
+        );
     }
 
     // Remove ** around bold text, replace with "
     function cleanBold(content: string): string {
-        const boldLiteral = /\*\*([\w|\s]+)\*\*/
-        const boldRegex = new RegExp(boldLiteral, "g");
-        const boldRemoved = content.replace(boldRegex, (matched, word, offset, s) => {
-            return "\"" + word + "\""
-        })
-        return boldRemoved;
+        return content.replace(
+            /\*\*([^*]+)\*\*/gu,
+            (matched, word, offset, s) => lf("\"{0}\"", word)
+        );
+    }
+
+    // replace horizontal rules with new lines
+    function cleanHorizontalRule(content: string): string {
+        return content.replace(
+            /^\* \* \*$/gum,
+            "- - -"
+        );
+    }
+
+    // replace new lines with break tags so that they don't get smooshed
+    function cleanNewLines(content: string): string {
```

Comment:
I don't really get this rule >> I've noticed it making breaks at the end of lines -- 

like 
![image](https://user-images.githubusercontent.com/18712271/112030667-bac59c00-8af7-11eb-847c-fb3cce4b97e8.png)

became
![image](https://user-images.githubusercontent.com/18712271/112030704-c31dd700-8af7-11eb-8368-c61fb22544cd.png)

was this made for a different case?

Reply:
Oh hm good point, I did it because currently none of the new lines are going through so everything is a single line as far as I've seen; I guess would be better as '2+ consecutive new lines or new lines followed by a bullet point'?

Reply:
oooooooooooooooooooooooooh I see what you're saying. Yeah, 2+ new lines should cover it?